### PR TITLE
Get rid off typeid in symmetric_tensor_36

### PR DESCRIPTION
**Pull request by masterleinad**
`symmetric_tensor_36` failed because the return value of `typeid(std::complex<...>).name()` differs from what is stored in the output file. Since this is not the essential part of the test, I would just choose something human-readable.

### CODE REVIEW

File: tests/base/symmetric_tensor_36.output [hunk 1]
```diff
@@ -1,98 +1,98 @@
 
-DEAL::Rank 2, Dim 2:  From f  To f ... OK
-DEAL::Rank 2, Dim 2:  From d  To f ... OK
-DEAL::Rank 2, Dim 2:  From f  To d ... OK
-DEAL::Rank 2, Dim 2:  From d  To d ... OK
-DEAL::Rank 2, Dim 2:  From f  To NSt3__17complexIfEE ... OK
```
Comment (jppelteret) [inline tests/base/symmetric_tensor_36.output]: The result of transposing the source and result number types is evident here.

File: tests/base/symmetric_tensor_36.cc [hunk 2]
```diff
@@ -38,21 +37,21 @@ template<int rank, int dim,
...
          typename NumberType1,
          typename NumberType2>
 typename boost::enable_if<boost::is_constructible<NumberType1,NumberType2>,void>::type
-test_tensor_constructor ()
+test_tensor_constructor (const std::string &type1, const std::string &type2)
 {
   deallog
       << "Rank " << rank << ", "
       << "Dim " << dim << ":"
-      << "  From " << typeid(NumberType2).name()
```
Comment (jppelteret) [inline tests/base/symmetric_tensor_36.cc]: It looks like the source and destination number types have been transposed.

**Comment by masterleinad** [comment]
@jppelteret source and destination number types are in the right order again.

**Comment by jppelteret** [comment]
Thanks a lot @masterleinad !

**Pull request merged**